src/randgraphs.jl

```diff
@@ -529,9 +529,9 @@ function stochastic_block_model{T<:Real}(c::Matrix{T}, n::Vector{Int}; seed::Int
         for b=a:K
             @assert a==b? c[a,b] <= n[b]-1 : c[a,b] <= n[b]   "Mean degree cannot be greater than available neighbors in the block."
 
-            m = a==b ? n[a]*(n[a]-1)/2 : n[a]*n[b]
+            m = Int(a==b ? n[a]*(n[a]-1)/2 : n[a]*n[b])
```

Comment:
It would be better to use integer division. Then the elements would never be promoted to floats.